event/batch_test.go

```diff
@@ -0,0 +1,200 @@
+package event_test
+
+import (
+	"github.com/ONSdigital/dp-observation-importer/event"
+	"github.com/ONSdigital/dp-observation-importer/schema"
+	"github.com/ONSdigital/go-ns/kafka/kafkatest"
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+)
+
+func TestIsEmpty(t *testing.T) {
+
+	Convey("Given a batch that is not empty", t, func() {
+
+		expectedEvent := getExampleEvent()
+		message := kafkatest.NewMessage([]byte(Marshal(*expectedEvent)))
+
+		batchSize := 1
+		batch := event.NewBatch(batchSize)
+
+		Convey("When the batch has no messages added", func() {
+			Convey("The batch is empty", func() {
+				So(batch.IsEmpty(), ShouldBeTrue)
+			})
+		})
+
+		Convey("When the batch has a message added", func() {
+
+			batch.Add(message)
+
+			Convey("The batch is now empty", func() {
+				So(batch.IsEmpty(), ShouldBeFalse)
+			})
+		})
+	})
+}
+
+func TestAdd(t *testing.T) {
+
+	Convey("Given a batch", t, func() {
+
+		expectedEvent := getExampleEvent()
+		message := kafkatest.NewMessage([]byte(Marshal(*expectedEvent)))
+
+		batchSize := 1
+		batch := event.NewBatch(batchSize)
+
+		Convey("When add is called with a valid message", func() {
+
+			batch.Add(message)
+
+			Convey("The batch contains the expected event.", func() {
+				So(len(batch.Events()), ShouldEqual, 1)
```

Comment:
They should always be the same. I have added checks throughout to make sure that they are 😄